main.go

```diff
@@ -258,9 +265,26 @@ type tenant struct {
 	} `json:"rateLimits"`
 }
 
+func init() {
+	version.Version = Version
+	version.Branch = Branch
+	version.Revision = Revision
+}
+
 //nolint:funlen,gocyclo,gocognit
 func main() {
+
+	printVersion := flag.Bool("version", false, "Print this builds version information")
+
 	cfg, err := parseFlags()
+
+	stdlog.Println(version.Print(cfg.debug.name))
+
+	if *printVersion {
+		// Only prints the version and exit
+		os.Exit(0)
+	}
+
```

Comment:
Two suggestions below:
1. Use `version.Info()` for the default logging statement. This a good enough one-liner when running the container.
2. Move the flag parsing to `parseFlags`

```suggestion
	cfg, err := parseFlags()
	if err != nil {
		stdlog.Fatalf("parse flag: %v", err)
	}

	if cfg.printVersionOnly {
		stdlog.Println(version.Print(cfg.debug.name))
		os.Exit(0)
	} else {
		stdlog.Println(version.Info())
	}
```